sbi/neural_nets/estimators/base.py

```diff
@@ -443,6 +509,447 @@ def diffusion_fn(self, input: Tensor, times: Tensor) -> Tensor:
+        """
+        raise NotImplementedError("Std_t is not implemented for this estimator.")
+
+    def drift_fn(self, input: Tensor, times: Tensor) -> Tensor:
+        r"""Drift function :math:`f(t)` of the vector field estimator.
+
+        The drift function :math:`f(t)` and diffusion function :math:`\g(t)`
+        enable SDE sampling:
+
+        .. math::
+            d\theta_t = [f(t) - g(t)^2 \nabla_{\theta_t} \log p(\theta_t | x_o)]dt
+              + \g(t) dW_t
+
+        where :math:`dW_t` is the Wiener process.
+
+
+        Args:
+            input: input parameters :math:`\theta_t`.
+            times: SDE time variable in [0,1].
+
+        Raises:
+            NotImplementedError: Drift is not implemented for this estimator.
+
+        """
+        raise NotImplementedError("Drift is not implemented for this estimator.")
+
+    def diffusion_fn(self, input: Tensor, times: Tensor) -> Tensor:
+        r"""Diffusion function :math:`\g(t)` of the vector field estimator.
+
+        The drift function :math:`f(t)` and diffusion function :math:`\g(t)`
+        enable SDE sampling:
+
+        .. math::
+            d\theta_t = [f(t) - g(t)^2 \nabla_{\theta_t} \log p(\theta_t | x_o)]dt
+              + \g(t) dW_t
+
+        where :math:`dW_t` is the Wiener process.
+
+        Args:
+            input: input parameters :math:`\theta_t`.
+            times: SDE time variable in [0,1].
+
+        Raises:
+            NotImplementedError: Diffusion is not implemented for this estimator.
+        """
+        raise NotImplementedError("Diffusion is not implemented for this estimator.")
+
+
+class MaskedConditionalVectorFieldEstimatorWrapper(ConditionalVectorFieldEstimator):
+    def __init__(self, original_estimator, fixed_condition_mask, fixed_edge_mask):
+        T, F = original_estimator.input_shape
+
+        num_latent = int(torch.sum(fixed_condition_mask == 0).item())
+        num_observed = int(torch.sum(fixed_condition_mask == 1).item())
+
+        # Count number of latent and observed nodes
+        self._new_input_shape = torch.Size((num_latent * F,))
+        self._new_condition_shape = torch.Size((num_observed * F,))
+
+        super().__init__(
+            net=original_estimator.net,
+            input_shape=self._new_input_shape,
+            condition_shape=self._new_condition_shape,
+            t_min=original_estimator.t_min,
+            t_max=original_estimator.t_max,
+        )
+
+        self.SCORE_DEFINED = original_estimator.SCORE_DEFINED
+        self.SDE_DEFINED = original_estimator.SDE_DEFINED
+        self.MARGINALS_DEFINED = original_estimator.MARGINALS_DEFINED
+
+        self._original_T = T
+        self._original_F = F
+        self._num_latent = num_latent
+        self._num_observed = num_observed
+
+        self._original_estimator = original_estimator
+
+        # Ensure input_part and condition_part are on the same device
+        device = next(original_estimator.net.parameters()).device
```

Comment:
This should be unnecessary. If you register them as buffers as below, they should be moved to the right device.